docs/development/pigs/pig-028.rst

```diff
@@ -0,0 +1,285 @@
+
+
+
+
+Implementation impact
+==============
+
+The implementation of the Bayesian inference class implies some addition/modifications to the current codebase.
+
+Namely: 
+
+1. Likelihood: The `stat_sum` value needs to be separated into a likelihood term and a prior penalty term.
+   While prior will be attached to the model, we want the stat_sum to return only the likelihood and not the penalty term as it is not needed
+   for the NS technique. In addition, the NS alogrithm aims to maximize the likelihood (not minimize) and the factor of 2 for the Cstat needs to be removed.
+   So likelihood should be : `-0.5*Stat`
+
+2. Priors: The sampling in the NS algorithm is drawn from a "unity cube". This means that all sample are drawn from a [0,1] random uniform distribution that 
+   are latter transformed to the parameter space via their inverse cumulative distribution (cdf). An `inverse_cdf` attribute needs to be implemented for all Priors.
+
+3. Sampler: The UltraNest sampler object needs to be initiliazed with the correct parameters and (modified) likelihood 
+
+
+
+Outputs from the Nested Sampling 
+==============
+
+Similarly to the MCMC method, the NS sampler will return samples of the posterior distributions. However there are several conceptual differences with the chains returned by the MCMC sampler.
+
+1. There is no concept of chains in NS, each sample (live or dead points) serve to explore the entire likelihood landscape. 
+   To estimate the posterior distributions, the samples need to be weighted in NS (See review paper listed in the intro). 
+2. The NS computes the full integral of the likelihood over the entire prior space. The integral result is the Bayesian evidence Z wich can be used to compare different models in apparently a better way that the BIC/AIC.
+   In addition an error on Z is estimated via bootstrap of the samples.
+3. A `converged : True/False` status can be returned
+   
+The UltraNest sampler returns a dictionnary that is very complete and that contains all (and even more) information that is needed.  
+Here are the keys of that dictionnary :   
+
+.. code:: python
+
+  ['niter', 'logz', 'logzerr', 'logz_bs', 'logz_single', 'logzerr_tail', 'logzerr_bs', 'ess', 'H', 'Herr', 'posterior', 'weighted_samples', 'samples', 'maximum_likelihood', 'ncall', 'paramnames', 'logzerr_single', 'insertion_order_MWW_test'])
+
+
+In particular, the posterior key returns all the information needed to update the model Parameters :   
+
+.. code:: python
+
+ 'posterior': {'mean': [2.014811633859834, 2.989779066475214e-12],
+  'stdev': [0.010066705375548107, 3.739770898783367e-14],
+  'median': [2.0148015517529894, 2.988413141629203e-12],
+  'errlo': [2.00472179200096, 2.9537056583888585e-12],
+  'errup': [2.024897178551308, 3.028549898215494e-12],
+
+
+
+Proposal
+==============
+
+
+
+.. code:: python
+
+  class NestedSampler:
+    def __init__(self, live_points=100, frac_remain=0.1):
```

Comment:
You need to be careful whether you want samples or weighted samples in this case.  I think samples is the good one